cmd/nitro/init.go

```diff
@@ -545,7 +548,50 @@ func openInitializeChainDb(ctx context.Context, stack *node.Node, config *NodeCo
 		if config.Init.ThenQuit {
 			cacheConfig.SnapshotWait = true
 		}
-		l2BlockChain, err = execution.WriteOrTestBlockChain(chainDb, cacheConfig, initDataReader, chainConfig, config.Node.TxLookupLimit, config.Init.AccountsPerSync)
+		var serializedChainConfig []byte
+		if config.Node.L1Reader.Enable {
+			delayedBridge, err := arbnode.NewDelayedBridge(l1Client, rollupAddrs.Bridge, rollupAddrs.DeployedAt)
+			if err != nil {
+				return chainDb, nil, fmt.Errorf("failed creating delayed bridge while attempting to get serialized chain config from init message: %w", err)
+			}
+			deployedAt := new(big.Int).SetUint64(rollupAddrs.DeployedAt)
+			delayedMessages, err := delayedBridge.LookupMessagesInRange(ctx, deployedAt, deployedAt, nil)
+			if err != nil {
+				return chainDb, nil, fmt.Errorf("failed getting delayed messages while attempting to get serialized chain config from init message: %w", err)
+			}
+			var initMessage *arbostypes.L1IncomingMessage
+			for _, msg := range delayedMessages {
+				if msg.Message.Header.Kind == arbostypes.L1MessageType_Initialize {
+					initMessage = msg.Message
+					break
+				}
+			}
+			if initMessage == nil {
+				return chainDb, nil, fmt.Errorf("failed to get init message while attempting to get serialized chain config")
+			}
+			var initChainConfig *params.ChainConfig
+			var initChainId *big.Int
+			initChainId, initChainConfig, serializedChainConfig, err = initMessage.ParseInitMessage()
+			if err != nil {
+				return chainDb, nil, err
+			}
+			if initChainId.Cmp(chainId) != 0 {
+				return chainDb, nil, fmt.Errorf("expected L2 chain ID %v but read L2 chain ID %v from init message in L1 inbox", chainId, initChainId)
+			}
+			if initChainConfig != nil {
+				if err := initChainConfig.CheckCompatible(chainConfig, chainConfig.ArbitrumChainParams.GenesisBlockNum, 0); err != nil {
+					return chainDb, nil, fmt.Errorf("incompatible chain config read from init message in L1 inbox: %w", err)
+				}
+			}
+			log.Info("Read serialized chain config from init message", "json", string(serializedChainConfig))
+		} else {
+			serializedChainConfig, err = json.Marshal(chainConfig)
```

Comment:
Since we're doing a `log.Info` in the other case I think it'd make sense to log here that we've serialized the chain config because the L1 reader was disabled. Maybe even a `log.Warn` because likely the user should have the L1 reader enabled.